src/classes/ATTD_CourseConnectionContact_TDTM.cls

```diff
@@ -0,0 +1,93 @@
+    
+    * Redistributions of source code must retain the above copyright
+      notice, this list of conditions and the following disclaimer.
+    * Redistributions in binary form must reproduce the above copyright
+      notice, this list of conditions and the following disclaimer in the
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of Salesforce.org nor the names of
+      its contributors may be used to endorse or promote products derived
+      from this software without specific prior written permission.
+ 
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT 
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS 
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE 
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, 
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, 
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; 
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER 
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT 
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN 
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE 
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2019
+* @group Attendance Events
+* @group-content ../../ApexDocContent/AttendanceEvents.htm
+* @description Makes sure that linked course connection's contact matches contact lookup field
+*/
+public with sharing class ATTD_CourseConnectionContact_TDTM extends TDTM_Runnable {
+    /*******************************************************************************************************
+    * @description Makes sure that linked course connection's contact matches contact lookup field
+    * @param listNew the list of Attendance Events from trigger new. 
+    * @param listOld the list of Attendance Events from trigger old. 
+    * @param triggerAction which trigger event (BeforeInsert, AfterInsert, etc.). 
+    * @param objResult the describe for Attendance Events 
+    * @return dmlWrapper.  
+    ********************************************************************************************************/
+    public override DmlWrapper run(List<SObject> newlist, List<SObject> oldlist, 
+    TDTM_Runnable.Action triggerAction, Schema.DescribeSObjectResult objResult) {
+
+        DmlWrapper dmlWrapper = new DmlWrapper();
+
+        boolean recursionFlag = TDTM_ProcessControl.getRecursionFlag(TDTM_ProcessControl.registeredTrigger.ATTD_CourseConnectionContact_TDTM);
+
+        Map<ID, SObject> newmap = new Map<ID, SObject>(newlist);
+        
+        if (triggerAction == TDTM_Runnable.Action.AfterInsert
+                || triggerAction == TDTM_Runnable.Action.AfterUpdate) {
+                    
+            if (!recursionFlag) {
```

Comment:
There's a lot of nested if statements here - can we check if recursionFlag is true after line 51 and if it is, just exit this method entirely?

Reply:
Well I don't think it will run those nested if statements if recursionflag is true anway

Reply:
That's fine, I was just hoping we could get rid of so many nested if statements here. I'm a big fan of exiting earlier and the ApexPMD plugin would probably flag this for having too many nested levels, but for sake of getting this through I'm okay with leaving this as-is.